static/app/views/starfish/queries/useReleases.tsx

```diff
@@ -39,3 +43,59 @@ export function useReleaseSelection() {
 
   return {primaryRelease, secondaryRelease, isLoading};
 }
+
+export function useReleaseStats() {
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const {environments, projects} = selection;
+
+  const {start, end, statsPeriod} = normalizeDateTimeParams(selection.datetime, {
+    allowEmptyPeriod: true,
+  });
+
+  // The sessions endpoint does not support wildcard search.
+  // So we're just getting top 250 values ordered by count.
+  // Hopefully this is enough to populate session count for
+  // any releases searched in the release selector.
+  const urlQuery = Object.fromEntries(
+    Object.entries({
+      project: projects,
+      environment: environments,
+      field: ['sum(session)'],
+      groupBy: ['release', 'project'],
+      orderBy: '-sum(session)',
+      start,
+      end,
+      statsPeriod,
+      per_page: 250,
+      interval: getInterval({start, end, period: statsPeriod}, 'low'),
+    }).filter(([, value]) => defined(value) && value !== '')
+  );
+
+  const {data, isLoading} = useApiQuery<any>(
+    [
+      `/organizations/${organization.slug}/sessions/`,
+      {
+        query: urlQuery,
+      },
+    ],
+    {staleTime: Infinity}
+  );
+
+  const releaseStatsMap: {[release: string]: {project: number; 'sum(session)': number}} =
+    {};
+  if (data && data.groups) {
+    data.groups.forEach(group => {
+      const release = group.by.release;
+      const project = group.by.project;
+      const sessionCount = group.totals['sum(session)'];
+
+      releaseStatsMap[release] = {project, 'sum(session)': sessionCount};
+    });
+  }
+
+  return {
+    data: releaseStatsMap,
+    isLoading,
+  };
```

Comment:
A pattern I like is returning the entire result from the api query, so u don't loose other properties like `isFetching`.
```suggestion
  const result = useApiQuery<any>(
    [
      `/organizations/${organization.slug}/sessions/`,
      {
        query: urlQuery,
      },
    ],
    {staleTime: Infinity}
  );

  const releaseStatsMap: {[release: string]: {project: number; 'sum(session)': number}} =
    {};
  if (result.data && result.data.groups) {
    result.data.groups.forEach(group => {
      const release = group.by.release;
      const project = group.by.project;
      const sessionCount = group.totals['sum(session)'];

      releaseStatsMap[release] = {project, 'sum(session)': sessionCount};
    });
  }

  return {
    ...result,
    data: releaseStatsMap,
  };
```

Reply:
good suggestion!